src/main.rs

```diff
@@ -505,6 +509,23 @@ fn render_templates(site_data: &SiteData, tera: &Tera, output_dir: &Path) -> Res
     Ok(())
 }
 
+fn render_404_page(content_dir: &Path) -> String {
+    let path_404_md = content_dir.join("404.md");
+
+    match path_404_md.exists() {
+        true => {
+            let markdown_content = fs::read_to_string(&path_404_md)
+                .expect("Failed to read the 404.md file");
+
+            let options = ComrakOptions::default();
+            markdown_to_html(&markdown_content, &options)
+        },
+        false => String::from("<h1>404 - Page Not Found</h1>"),
+    }
+}
```

Comment:
Then this function can return nothing (or a Result<> if you want to handle error)

```suggestion
fn render_404_page(content_dir: &Path, output_dir: &Path) {
    let path_404_md = content_dir.join("404.md");
    if !path_404_md.exists() {
        let content = String::from("<h1>404 - Page Not Found</h1>");
        let mut file = File::create("output.txt");
        file.write_all(content.as_bytes());
    }
}
``